src/components/generics/vacation-test-forms/employee-vacation-requests.tsx

```diff
@@ -92,6 +105,68 @@ const RenderEmployeeVacationRequests = ({ persons }: Props) => {
     initializeRequests();
   }, [person]);
 
+  /**
+   * Initializes all vacation request statuses
+   */
+  const initializeRequestStatuses = async () => {
+    try {
+      const vacationRequestStatuses: VacationRequestStatus[] = [];
+      const statusesApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+
+      await Promise.all(requests.map(async request => {
+        const createdStatuses = await statusesApi.listVacationRequestStatuses({ id: request.id! });
```

Comment:
It's best to avoid the type assertions where possible e.g. request.id!, one way around this would be to do a conditional check (within the map) to check request.id is defined, and if not you could return out or throw an error- whichever you think most appropriate

Reply:
Would this error handling suffice?

```suggestion
    let createdStatuses: VacationRequestStatus[];
    if (request.id) {
      createdStatuses = await statusesApi.listVacationRequestStatuses({ id: request.id });
    } else {
      throw new Error("Vacation Request ID undefined!");
    }
```